tests/sentry/api/endpoints/test_organization_releases.py

```diff
@@ -891,3 +891,150 @@ def test_bad_repo_name(self):
+            organization_id=org.id,
+            project_id=project1.id,
+            release_id=release1.id,
+            environment_id=env1.id,
+        )
+
+        release2 = Release.objects.create(
+            organization_id=org.id,
+            version='2',
+            date_added=datetime(2013, 8, 14, 3, 8, 24, 880386),
+        )
+        release2.add_project(project2)
+        ReleaseEnvironment.objects.create(
+            organization_id=org.id,
+            project_id=project2.id,
+            release_id=release2.id,
+            environment_id=env2.id,
+        )
+
+        release3 = Release.objects.create(
+            organization_id=org.id,
+            version='3',
+            date_added=datetime(2013, 8, 12, 3, 8, 24, 880386),
+            date_released=datetime(2013, 8, 15, 3, 8, 24, 880386),
+        )
+        release3.add_project(project1)
+        ReleaseEnvironment.objects.create(
+            organization_id=org.id,
+            project_id=project1.id,
+            release_id=release3.id,
+            environment_id=env2.id,
+        )
+
+        release4 = Release.objects.create(
+            organization_id=org.id,
+            version='4',
+        )
+        release4.add_project(project2)
+
+        self.project1 = project1
+        self.project2 = project2
+
+        self.release1 = release1
+        self.release2 = release2
+        self.release3 = release3
+        self.release4 = release4
+
+        self.env1 = env1
+        self.env2 = env2
+        self.org = org
+
+    def make_environment(self, name, project):
+        env = Environment.objects.create(
+            project_id=project.id,
+            organization_id=project.organization_id,
+            name=name,
+        )
+        env.add_project(project)
+        return env
+
+    def assert_releases(self, response, releases):
+        assert response.status_code == 200, response.content
+        assert len(response.data) == len(releases)
+
+        response_versions = sorted([r['version'] for r in response.data])
+        releases_versions = sorted([r.version for r in releases])
+        assert response_versions == releases_versions
+
+    def test_environments_filter(self):
+        url = reverse(
+            'sentry-api-0-organization-releases',
+            kwargs={
+                'organization_slug': self.org.slug,
+            }
+        )
+        response = self.client.get(url + '?environment=' + self.env1.name, format='json')
+        self.assert_releases(response, [self.release1])
+
+        response = self.client.get(url + '?environment=' + self.env2.name, format='json')
+        self.assert_releases(response, [self.release2, self.release3])
+
+    def test_no_environment(self):
+        url = reverse(
+            'sentry-api-0-organization-releases',
+            kwargs={
+                'organization_slug': self.org.slug,
+            }
+        )
+        response = self.client.get(url + '?environment=', format='json')
```

Comment:
@tkaemming I'm fine with pulling that test out. You're right it does feel redundant since I added test_invalid_environment. The only thing I was noticing that `test_empty_environment` would return the same result if that empty environment hadn't been created first. I would argue that's worth noting and having it documented with a test.

Reply:
I found it an odd behavior that I wouldn't have expected to be honest. So I wrote a test around it.